lib/stages/DotProduct.cpp

```diff
@@ -0,0 +1,107 @@
+#include "DotProduct.hpp"
+
+#include "StageFactory.hpp"
+#include "errors.h"
+#include "visUtil.hpp" // for frameID, modulo
+
+/* Load the classes we will be using
+ * TODO: Annotate the includes and usings */
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+
+// Register the stage with abstract factory.
+// Needed for Kotekan to be able to find the stage.
+REGISTER_KOTEKAN_STAGE(DotProduct);
+
+/* Constructor for the stage
+    TODO: Annotate the arguments everywhere
+    Note, that you can use the macro STAGE_CONSTRUCTOR(DotProduct)
```

Comment:
I would use the `STAGE_CONSTRUCTOR` in one of the two example stages, so people can see it in action.